### PR TITLE
Chore: discard trailing coma from JSON settings files

### PR DESCRIPTION
**Pull request by alexduros**
When JSON VSCode configurations files contains trailing comma - which seems to be accepted by VSCode, it is not accepted by the parser from the extension causing failure in task launches. This was working in previous versions.

Note: I'm sorry for the changes but while following the steps to lint properly my PR, I'm still getting indent changes from the linting process (VSCode shows me that the tslint version for this project is 3.9.7). I don't see how to clean it up properly.

### CODE REVIEW

**Comment by SounD120** [comment]
Hi @alexduros and thanks for your contribution! Looks interesting, but could you please also add some unit tests in order to test your RegExp on real cases?

**Comment by alexduros** [comment]
Thanks @SounD120 ! I added the nominal test case (tbh, I was a bit struggling with having a working testing development environment 👍 ) 
On the top of that, I'm still missing how to fit to your linting rules. My PR is a bit messing with your codebase in the current state.

**Comment by SounD120** [comment]
@alexduros What caused the style changes? As seen from your PR there are not so much changes, but for some reason a lot of the code was reformatted. Have you launch `eslint` formatter?

**Comment by alexduros** [comment]
My bad, it's my Prettier extension I forgot to turn off which was messing everything. Let me cleanup that.

**Comment by alexduros** [comment]
This should be better (and clearer), now @SounD120

**Comment by SounD120** [comment]
Hi @alexduros , we tested your RegExp and it seems that it might cause breakage in some cases like this:
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug Android",
      "cwd": "${workspaceFolder}",
      "type": "reactnative",
      "request": "launch",
      "platform": "android",
      "logCatArguments": ["ReactNative", "ReactNativeJS"],
      "env": {
        "testvar": "(value0), (value1)"
      }
    }
  ]
}
```
After this RegExp `testvar` in `env` will become `(value0) (value1)` instead of `(value0), (value1)`. Could you please take a look?

**Comment by alexduros** [comment]
Indeed, @SounD120 . Improved it a bit with the use case added to unit tests.

File: test/common/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                {
+                  "name": "Debug Android",
+                  "cwd": "\${workspaceFolder\}",
+                  "type": "reactnative",
+                  "request": "launch",
+                  "platform": "android",
+                  "logCatArguments": ["ReactNative", "ReactNativeJS"],
+                  "env": {
+                    "testvar": "(value0), (value1), (value2)"
+                  }
```
Comment (SounD120) [inline test/common/utils.test.ts:53]:
```suggestion
                  },
```

**Review by RedMickey — COMMENTED** [review]
Hi @alexduros, I tested your RegExp and found out a case when it could work incorrectly:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug Android",
            "cwd": "${workspaceFolder}",
            "type": "reactnative",
            "request": "launch",
            "platform": "android"
        }
    ]
},
```
Screen Shot 2020-10-12 at 16 53 29

The RegExp skips the last trailing comma (after the last curly brace). Could you please take a look on this?

**Comment by alexduros** [comment]
Hey @RedMickey, sorry I completely missed that 😖 . 
Actually, I fixed it in a very simple and dumb way. Tbh, I felt this more as a retrocompatibility feat. more than a real JSON sanitizer (that's, I didn't go for a third-party and a very basic regex impl.).
Let me know 😉

**Comment by SounD120** [comment]
Hi @alexduros . Yes, I think it makes sense to sanitize JSON using a lib.

**Comment by etatanova** [comment]
Hi @alexduros, we've tested your RegExp and it seems that trailing commas are discarded even in quoted expressions. We may suggest using [JSON5](https://github.com/json5/json5) package since

- `JSON5.parse()` supports trailing commas;
- it allows to have single and multi-line comments in a document.

So we propose to change code this way:

- remove [using](https://github.com/microsoft/vscode-react-native/pull/1422/files#diff-e132e2d58d8a919473d09c7a11e16a97af01c1dae64e1fcbbcc821436678d3f8R323) of the `strip-json-comments` package;
- [add](https://github.com/json5/json5#installation)  `json5` package to the project;
- leave the regular expression `endOfStringTrailingCommaRegex` and use it before `JSON5.parse()` call  since JSON5 doesn’t support the case when the comma is in the end of JSON;
- call `JSON5.parse()` instead of `JSON.parse()`, so a JSON document should be parsed correctly.

Could you please take a look at this?

**Comment by alexduros** [comment]
Indeed. Clearly, as previously my attempt was really not to fully support this feat. (and hence not relying to a third party). 
I will take a look 👍

**Comment by etatanova** [comment]
Hi @alexduros, do you have a chance to look at this? Do you need any help from our side?

**Comment by etatanova** [comment]
Hi @alexduros, do you have any updates?

**Comment by etatanova** [comment]
Hi @alexduros, do you have a chance to look at this?

**Comment by EzioLi01** [comment]
Hey @alexduros, do you have chance to look into this?

**Review by EzioLi01 — APPROVED** [review]
LGTM!

**Comment by alexduros** [comment]
Oh, thank your @EzioLi01 @jinquan2019, completely missed that. 🙏🏻 🙈

**Comment by EzioLi01** [comment]
@alexduros NP, thanks for the contribution to project!

**Pull request merged**